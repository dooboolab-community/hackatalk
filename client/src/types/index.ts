import { StyleProp, TextStyle } from 'react-native';

import { SFC } from 'react';
import { User } from './graphql';

export enum MessageType {
  Message,
  Photo,
  File,
}

export interface Friend extends User {
  isFriend?: boolean;
  friendSince: Date;
}

interface IconProps {
  style?: StyleProp<TextStyle>;
  width?: number | string;
  height?: number | string;
  children?: never;
}

export type IconType = SFC<IconProps>;

export enum ThemeType {
  LIGHT = 'LIGHT',
  DARK = 'DARK',
}
