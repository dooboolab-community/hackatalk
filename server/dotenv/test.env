JWT_SECRET=dooboolab
DATABASE_URL="postgresql://test:test!@localhost:5432/postgres?schema=test"
