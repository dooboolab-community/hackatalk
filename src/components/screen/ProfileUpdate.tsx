import React, { useEffect, useState } from 'react';
import { TouchableOpacity, View } from 'react-native';

import Button from '../shared/Button';
// import { CommonActions } from '@react-navigation/core';
import { DefaultNavigationProps } from '../../types';
import { Ionicons } from '@expo/vector-icons';
import TextInput from '../shared/TextInput';
import { getString } from '../../../STRINGS';
import styled from 'styled-components/native';
import { useThemeContext } from '../../providers/ThemeProvider';

const StyledContainer = styled.View`
  flex: 1;
  background-color: ${({ theme }): string => theme.background};
  flex-direction: column;
  align-items: center;
`;

const StyledScrollView = styled.ScrollView`
  align-self: stretch;
`;

const StyledWrapper = styled.View`
  margin-top: 48px;
  width: 78%;
  flex-direction: column;
  align-items: center;
`;

const StyledBtnWrapper = styled.View`
  flex: 1;
  justify-content: space-between;
  flex-direction: row;
  align-items: center;
  align-self: stretch;
  height: 60px;
  margin-top: 20px;
  margin-bottom: 48px;
`;

interface Props {
  navigation: DefaultNavigationProps;
}

function Screen(props: Props): React.ReactElement {
  const [isUpdating, setIsUpdating] = useState(false);
  const [displayName, setDisplayName] = useState('');
  const [statusMsg, setStatusMsg] = useState('');

  useEffect(() => {
    if (isUpdating) {
      try {
        if (props.navigation) {
          props.navigation.goBack();
        }
      } catch (err) {
        // console.error(err);
      } finally {
        setIsUpdating(false);
      }
    }
  }, [isUpdating]);

  const onLogout = (): void => {
    if (props.navigation) {
      props.navigation.resetRoot({
        index: 0,
        routes: [{ name: 'AuthStack' }],
      });
    }
  };

  const onUpdate = (): void => {
    setIsUpdating(true);
  };

  const onTextChanged = (type: string, text: string): void => {
    // prettier-ignore
    switch (type) {
      case 'DISPLAY_NAME':
        setDisplayName(text);
        break;
      case 'STATUS_MSG':
        setStatusMsg(text);
        break;
    }
  };

  const onPressImg = (): void => {};

  const { theme } = useThemeContext();

  return (
    <StyledContainer>
      <StyledScrollView
        contentContainerStyle={{
          justifyContent: 'center',
          alignItems: 'center',
        }}
      >
        <StyledWrapper>
          <TouchableOpacity activeOpacity={0.5} onPress={onPressImg}>
            <Ionicons
              name="ios-person"
              size={80}
              color={theme ? theme.fontColor : '#3d3d3d'}
            />
          </TouchableOpacity>
          <TextInput
            testID="input_name"
            style={{ marginTop: 80 }}
            txtLabel={getString('NAME')}
            txtHint={getString('NAME')}
            txt={displayName}
            onTextChanged={(text: string): void =>
              onTextChanged('DISPLAY_NAME', text)
            }
          />
          <TextInput
            testID="input_status"
            style={{ marginTop: 24 }}
            txtLabel={getString('STATUS_MSG')}
            txtHint={getString('STATUS_MSG')}
            txt={statusMsg}
            onTextChanged={(text: string): void =>
              onTextChanged('STATUS_MSG', text)
            }
          />
          <StyledBtnWrapper>
            <Button
              testID="logout_btn"
              onPress={onLogout}
              isWhite
              containerStyle={{ flex: 1, flexDirection: 'row' }}
            >
              {getString('LOGOUT')}
            </Button>
            <View style={{ width: 8 }} />
            <Button
              testID="update_btn"
              isLoading={isUpdating}
              onPress={onUpdate}
              containerStyle={{ flex: 1, flexDirection: 'row' }}
            >
              {getString('UPDATE')}
            </Button>
          </StyledBtnWrapper>
        </StyledWrapper>
      </StyledScrollView>
    </StyledContainer>
  );
}

export default Screen;
