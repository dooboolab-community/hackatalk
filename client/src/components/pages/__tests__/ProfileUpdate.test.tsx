import 'react-native';

import {MockPayloadGenerator, createMockEnvironment} from 'relay-test-utils';
import {fireEvent, render} from '@testing-library/react-native';

import ProfileUpdate from '../ProfileUpdate';
import React from 'react';
import {User} from '../../../types/graphql';
import {act} from 'react-test-renderer';
import {createTestElement} from '../../../../test/testUtils';

jest.mock('@expo/react-native-action-sheet', () => ({
  useActionSheet: (): Record<string, unknown> => {
    const userPressLaunchCamera = true;
    const BUTTON_INDEX_LAUNCH_CAMERA = 0;
    const BUTTON_INDEX_LAUNCH_IMAGE_LIBRARY = 1;

    return {
      showActionSheetWithOptions: (
        options: Record<string, unknown>,
        callback: (index: number) => void,
      ): void => {
        if (userPressLaunchCamera) callback(BUTTON_INDEX_LAUNCH_CAMERA);

        callback(BUTTON_INDEX_LAUNCH_IMAGE_LIBRARY);
      },
    };
  },
}));

jest.mock('expo-permissions', () => ({
  askAsync: (): {status: string} => ({
    status: 'granted',
  }),
}));

jest.mock('expo-image-picker', () => ({
  launchCameraAsync: (): string => 'photo info',
  launchImageLibraryAsync: (): string => 'photo info',
}));

const mockEnvironment = createMockEnvironment();

const generateUser = (_: unknown, generateId: () => number): User => ({
  id: `test-user-${generateId()}`,
  email: 'email',
  name: 'name',
  nickname: 'nickname',
  statusMessage: '',
  verified: true,
});

describe('rendering test', () => {
  it('renders as expected', () => {
    const component = createTestElement(<ProfileUpdate />);
    const screen = render(component);
    const json = screen.toJSON();

    expect(json).toBeTruthy();
    expect(json).toMatchSnapshot();
  });
});

describe('[ProfileUpdate] interaction', () => {
  it('should change nickname', async () => {
    const component = createTestElement(<ProfileUpdate />);
    const screen = render(component);
    const inputNickname = screen.getByTestId('input-nickname');

    fireEvent.changeText(inputNickname, 'nickname');

    expect(inputNickname.props.value).toEqual('nickname');
  });

  it('should change name', async () => {
    const component = createTestElement(<ProfileUpdate />);
    const screen = render(component);
    const inputName = screen.getByTestId('input-name');

    fireEvent.changeText(inputName, 'name');

    expect(inputName.props.value).toEqual('name');
  });

  it('should change status text', async () => {
    const component = createTestElement(<ProfileUpdate />);
    const screen = render(component);
    const inputStatus = screen.getByTestId('input-status');

    fireEvent.changeText(inputStatus, 'status');

    expect(inputStatus.props.value).toEqual('status');
  });

  it('should call updateProfile when update button pressed', async () => {
    const component = createTestElement(<ProfileUpdate />, {
      environment: mockEnvironment,
    });

    const screen = render(component);
    const updateButton = screen.getByTestId('button-update');

    fireEvent.press(updateButton);

    const operation = mockEnvironment.mock.getMostRecentOperation();

    act(() => {
      mockEnvironment.mock.resolve(
        operation,
        MockPayloadGenerator.generate(operation, {
          User: generateUser,
        }),
      );
    });
  });

  // it('should launch camera when user select "Take a picture"', async () => {
  //   // userPressLaunchCamera = true;
  //   const profileBtn = await waitForElement(() =>
  //     testingLib.queryByTestId('button-user-icon'),
  //   );
  //   await act(() => {
  //     fireEvent.press(profileBtn);
  //     // jest.runAllTimers();
  //   });
  //   await wait();
  // });

  // it('should open album when user select "Select from Album"', async () => {
  //   // userPressLaunchCamera = false;
  //   const profileBtn = await waitForElement(() =>
  //     testingLib.queryByTestId('button-user-icon'),
  //   );
  //   await wait(() => expect(profileBtn).toBeTruthy());
  //   act(() => {
  //     fireEvent.press(profileBtn);
  //     // jest.runAllTimers();
  //   });
  //   await wait();
  // });
});
