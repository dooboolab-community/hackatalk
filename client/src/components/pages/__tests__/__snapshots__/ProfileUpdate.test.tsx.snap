// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`rendering test renders as expected 1`] = `
<RNCSafeAreaProvider
  onInsetsChange={[Function]}
  style={
    [
      {
        "flex": 1,
      },
      undefined,
    ]
  }
>
  <View
    style={
      [
        {
          "alignItems": "center",
          "backgroundColor": "#000",
          "flexBasis": 0,
          "flexDirection": "column",
          "flexGrow": 1,
          "flexShrink": 1,
        },
        undefined,
      ]
    }
  >
    <RCTScrollView
      contentContainerStyle={
        {
          "alignItems": "center",
          "justifyContent": "center",
        }
      }
      style={
        [
          {
            "alignSelf": "stretch",
          },
          undefined,
        ]
      }
    >
      <View>
        <View
          style={
            [
              {
                "alignItems": "center",
                "flexDirection": "column",
                "marginTop": 48,
                "width": "78%",
              },
              undefined,
            ]
          }
        >
          <View
            accessible={true}
            collapsable={false}
            focusable={true}
            onClick={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              {
                "opacity": 1,
              }
            }
            testID="button-user-icon"
          >
            <View
              style={
                {
                  "height": 90,
                  "width": 90,
                }
              }
            >
              <Image
                source={
                  {
                    "canInstrument": true,
                    "createTransformer": [Function],
                    "getCacheKey": [Function],
                    "process": [Function],
                  }
                }
                style={
                  {
                    "height": 80,
                    "width": 80,
                  }
                }
              />
              <Image
                source={
                  {
                    "canInstrument": true,
                    "createTransformer": [Function],
                    "getCacheKey": [Function],
                    "process": [Function],
                  }
                }
                style={
                  {
                    "bottom": 0,
                    "position": "absolute",
                    "right": 0,
                  }
                }
              />
            </View>
          </View>
          <View
            style={
              [
                {
                  "alignSelf": "stretch",
                  "flexDirection": "column",
                },
                {
                  "marginTop": 26,
                },
              ]
            }
            testID="container-id"
          >
            <View
              style={
                [
                  [
                    {
                      "alignSelf": "stretch",
                      "borderBottomColor": "white",
                      "borderBottomWidth": 1,
                      "flexDirection": "column",
                      "justifyContent": "space-between",
                    },
                    {
                      "borderColor": "white",
                    },
                  ],
                  false,
                  {
                    "borderColor": "white",
                  },
                  {
                    "borderBottomColor": "white",
                  },
                ]
              }
            >
              <Text
                style={
                  [
                    [
                      {
                        "color": "#6D6D6D",
                        "fontSize": 14,
                        "paddingHorizontal": 4,
                      },
                      undefined,
                    ],
                    false,
                    {},
                  ]
                }
              >
                Nickname
              </Text>
              <TextInput
                autoCapitalize="none"
                autoCorrect={false}
                editable={true}
                multiline={false}
                onBlur={[Function]}
                onChangeText={[Function]}
                onFocus={[Function]}
                placeholder="A Nickname can help others find you"
                placeholderTextColor="#6D6D6D"
                secureTextEntry={false}
                style={
                  [
                    [
                      {
                        "color": "white",
                        "fontSize": 14,
                        "paddingHorizontal": 10,
                        "paddingVertical": 10,
                      },
                      {
                        "color": "white",
                      },
                    ],
                    false,
                  ]
                }
                testID="input-nickname"
                value=""
              />
            </View>
            <View
              style={
                {
                  "flex": 1,
                  "flexDirection": "row",
                  "justifyContent": "space-between",
                }
              }
            >
              <Text
                style={
                  [
                    [
                      {
                        "color": "#FF002E",
                        "fontSize": 12,
                        "marginTop": 8,
                        "paddingHorizontal": 10,
                      },
                      undefined,
                    ],
                    {
                      "color": "#FF002E",
                      "flex": 1,
                    },
                  ]
                }
              />
            </View>
          </View>
          <View
            style={
              [
                {
                  "alignSelf": "stretch",
                  "flexDirection": "column",
                },
                {
                  "marginTop": 12,
                },
              ]
            }
            testID="container-id"
          >
            <View
              style={
                [
                  [
                    {
                      "alignSelf": "stretch",
                      "borderBottomColor": "white",
                      "borderBottomWidth": 1,
                      "flexDirection": "column",
                      "justifyContent": "space-between",
                    },
                    {
                      "borderColor": "white",
                    },
                  ],
                  false,
                  {
                    "borderColor": "white",
                  },
                  {
                    "borderBottomColor": "white",
                  },
                ]
              }
            >
              <Text
                style={
                  [
                    [
                      {
                        "color": "#6D6D6D",
                        "fontSize": 14,
                        "paddingHorizontal": 4,
                      },
                      undefined,
                    ],
                    false,
                    {},
                  ]
                }
              >
                Name
              </Text>
              <TextInput
                autoCapitalize="none"
                autoCorrect={false}
                editable={true}
                multiline={false}
                onBlur={[Function]}
                onChangeText={[Function]}
                onFocus={[Function]}
                placeholder="Please write your name"
                placeholderTextColor="#6D6D6D"
                secureTextEntry={false}
                style={
                  [
                    [
                      {
                        "color": "white",
                        "fontSize": 14,
                        "paddingHorizontal": 10,
                        "paddingVertical": 10,
                      },
                      {
                        "color": "white",
                      },
                    ],
                    false,
                  ]
                }
                testID="input-name"
                value=""
              />
            </View>
            <View
              style={
                {
                  "flex": 1,
                  "flexDirection": "row",
                  "justifyContent": "space-between",
                }
              }
            >
              <Text
                style={
                  [
                    [
                      {
                        "color": "#FF002E",
                        "fontSize": 12,
                        "marginTop": 8,
                        "paddingHorizontal": 10,
                      },
                      undefined,
                    ],
                    {
                      "color": "#FF002E",
                      "flex": 1,
                    },
                  ]
                }
              />
            </View>
          </View>
          <View
            style={
              [
                {
                  "alignSelf": "stretch",
                  "flexDirection": "column",
                },
                {
                  "marginTop": 18,
                },
              ]
            }
            testID="container-id"
          >
            <View
              style={
                [
                  [
                    {
                      "alignSelf": "stretch",
                      "borderBottomColor": "white",
                      "borderBottomWidth": 1,
                      "flexDirection": "column",
                      "justifyContent": "space-between",
                    },
                    {
                      "borderColor": "white",
                      "borderWidth": 1,
                      "paddingHorizontal": 8,
                      "paddingVertical": 12,
                    },
                  ],
                  false,
                  {
                    "borderColor": "white",
                  },
                  {
                    "borderBottomColor": "white",
                  },
                ]
              }
            >
              <Text
                style={
                  [
                    [
                      {
                        "color": "#6D6D6D",
                        "fontSize": 14,
                        "paddingHorizontal": 4,
                      },
                      undefined,
                    ],
                    false,
                    {},
                  ]
                }
              >
                Status message
              </Text>
              <TextInput
                autoCapitalize="none"
                autoCorrect={false}
                editable={true}
                maxLength={60}
                multiline={true}
                onBlur={[Function]}
                onChangeText={[Function]}
                onFocus={[Function]}
                placeholder="Please write your status message"
                placeholderTextColor="#6D6D6D"
                secureTextEntry={false}
                style={
                  [
                    [
                      {
                        "color": "white",
                        "fontSize": 14,
                        "paddingHorizontal": 10,
                        "paddingVertical": 10,
                      },
                      {
                        "color": "white",
                        "marginTop": 12,
                      },
                    ],
                    false,
                  ]
                }
                testID="input-status"
                value=""
              />
            </View>
            <View
              style={
                {
                  "flex": 1,
                  "flexDirection": "row",
                  "justifyContent": "space-between",
                }
              }
            >
              <Text
                style={
                  [
                    [
                      {
                        "color": "#FF002E",
                        "fontSize": 12,
                        "marginTop": 8,
                        "paddingHorizontal": 10,
                      },
                      undefined,
                    ],
                    {
                      "color": "#FF002E",
                      "flex": 1,
                    },
                  ]
                }
              />
              <Text
                style={
                  [
                    {
                      "color": "white",
                      "fontSize": 12,
                      "marginTop": 8,
                      "paddingHorizontal": 10,
                    },
                    undefined,
                  ]
                }
              >
                0/60
              </Text>
            </View>
          </View>
          <View
            style={
              [
                {
                  "alignSelf": "stretch",
                  "flexDirection": "column",
                },
                {
                  "marginTop": 12,
                },
              ]
            }
            testID="container-id"
          >
            <View
              style={
                [
                  [
                    {
                      "alignSelf": "stretch",
                      "borderBottomColor": "white",
                      "borderBottomWidth": 1,
                      "flexDirection": "column",
                      "justifyContent": "space-between",
                    },
                    {
                      "borderColor": "white",
                    },
                  ],
                  false,
                  {
                    "borderColor": "white",
                  },
                  {
                    "borderBottomColor": "white",
                  },
                ]
              }
            >
              <Text
                style={
                  [
                    [
                      {
                        "color": "#6D6D6D",
                        "fontSize": 14,
                        "paddingHorizontal": 4,
                      },
                      undefined,
                    ],
                    false,
                    {},
                  ]
                }
              >
                Organization
              </Text>
              <TextInput
                autoCapitalize="none"
                autoCorrect={false}
                editable={true}
                multiline={false}
                onBlur={[Function]}
                onChangeText={[Function]}
                onFocus={[Function]}
                placeholder="Your organization."
                placeholderTextColor="#6D6D6D"
                secureTextEntry={false}
                style={
                  [
                    [
                      {
                        "color": "white",
                        "fontSize": 14,
                        "paddingHorizontal": 10,
                        "paddingVertical": 10,
                      },
                      {
                        "color": "white",
                      },
                    ],
                    false,
                  ]
                }
                value=""
              />
            </View>
            <View
              style={
                {
                  "flex": 1,
                  "flexDirection": "row",
                  "justifyContent": "space-between",
                }
              }
            >
              <Text
                style={
                  [
                    [
                      {
                        "color": "#FF002E",
                        "fontSize": 12,
                        "marginTop": 8,
                        "paddingHorizontal": 10,
                      },
                      undefined,
                    ],
                    {
                      "color": "#FF002E",
                      "flex": 1,
                    },
                  ]
                }
              />
            </View>
          </View>
          <View
            style={
              [
                {
                  "alignSelf": "stretch",
                  "flexDirection": "column",
                },
                {
                  "marginTop": 18,
                },
              ]
            }
            testID="container-id"
          >
            <View
              style={
                [
                  [
                    {
                      "alignSelf": "stretch",
                      "borderBottomColor": "white",
                      "borderBottomWidth": 1,
                      "flexDirection": "column",
                      "justifyContent": "space-between",
                    },
                    {
                      "borderColor": "white",
                      "borderWidth": 1,
                      "paddingHorizontal": 8,
                      "paddingVertical": 12,
                    },
                  ],
                  false,
                  {
                    "borderColor": "white",
                  },
                  {
                    "borderBottomColor": "white",
                  },
                ]
              }
            >
              <Text
                style={
                  [
                    [
                      {
                        "color": "#6D6D6D",
                        "fontSize": 14,
                        "paddingHorizontal": 4,
                      },
                      undefined,
                    ],
                    false,
                    {},
                  ]
                }
              >
                About me
              </Text>
              <TextInput
                autoCapitalize="none"
                autoCorrect={false}
                editable={true}
                maxLength={300}
                multiline={true}
                onBlur={[Function]}
                onChangeText={[Function]}
                onFocus={[Function]}
                placeholder="Who are you?"
                placeholderTextColor="#6D6D6D"
                secureTextEntry={false}
                style={
                  [
                    [
                      {
                        "color": "white",
                        "fontSize": 14,
                        "paddingHorizontal": 10,
                        "paddingVertical": 10,
                      },
                      {
                        "color": "white",
                        "marginTop": 12,
                      },
                    ],
                    false,
                  ]
                }
                value=""
              />
            </View>
            <View
              style={
                {
                  "flex": 1,
                  "flexDirection": "row",
                  "justifyContent": "space-between",
                }
              }
            >
              <Text
                style={
                  [
                    [
                      {
                        "color": "#FF002E",
                        "fontSize": 12,
                        "marginTop": 8,
                        "paddingHorizontal": 10,
                      },
                      undefined,
                    ],
                    {
                      "color": "#FF002E",
                      "flex": 1,
                    },
                  ]
                }
              />
              <Text
                style={
                  [
                    {
                      "color": "white",
                      "fontSize": 12,
                      "marginTop": 8,
                      "paddingHorizontal": 10,
                    },
                    undefined,
                  ]
                }
              >
                0/300
              </Text>
            </View>
          </View>
          <View
            style={
              [
                {
                  "alignSelf": "stretch",
                  "flexDirection": "column",
                },
                {
                  "marginTop": 18,
                },
              ]
            }
            testID="container-id"
          >
            <View
              style={
                [
                  [
                    {
                      "alignSelf": "stretch",
                      "borderBottomColor": "white",
                      "borderBottomWidth": 1,
                      "flexDirection": "column",
                      "justifyContent": "space-between",
                    },
                    {
                      "borderColor": "white",
                      "borderWidth": 1,
                      "paddingHorizontal": 8,
                      "paddingVertical": 12,
                    },
                  ],
                  false,
                  {
                    "borderColor": "white",
                  },
                  {
                    "borderBottomColor": "white",
                  },
                ]
              }
            >
              <Text
                style={
                  [
                    [
                      {
                        "color": "#6D6D6D",
                        "fontSize": 14,
                        "paddingHorizontal": 4,
                      },
                      undefined,
                    ],
                    false,
                    {},
                  ]
                }
              >
                Projects
              </Text>
              <TextInput
                autoCapitalize="none"
                autoCorrect={false}
                editable={true}
                maxLength={300}
                multiline={true}
                onBlur={[Function]}
                onChangeText={[Function]}
                onFocus={[Function]}
                placeholder="What are you working on and what have you done in the past?"
                placeholderTextColor="#6D6D6D"
                secureTextEntry={false}
                style={
                  [
                    [
                      {
                        "color": "white",
                        "fontSize": 14,
                        "paddingHorizontal": 10,
                        "paddingVertical": 10,
                      },
                      {
                        "color": "white",
                        "marginTop": 12,
                      },
                    ],
                    false,
                  ]
                }
                value=""
              />
            </View>
            <View
              style={
                {
                  "flex": 1,
                  "flexDirection": "row",
                  "justifyContent": "space-between",
                }
              }
            >
              <Text
                style={
                  [
                    [
                      {
                        "color": "#FF002E",
                        "fontSize": 12,
                        "marginTop": 8,
                        "paddingHorizontal": 10,
                      },
                      undefined,
                    ],
                    {
                      "color": "#FF002E",
                      "flex": 1,
                    },
                  ]
                }
              />
              <Text
                style={
                  [
                    {
                      "color": "white",
                      "fontSize": 12,
                      "marginTop": 8,
                      "paddingHorizontal": 10,
                    },
                    undefined,
                  ]
                }
              >
                0/300
              </Text>
            </View>
          </View>
          <View
            style={
              [
                {
                  "alignSelf": "stretch",
                  "flexDirection": "column",
                },
                {
                  "marginTop": 18,
                },
              ]
            }
            testID="container-id"
          >
            <View
              style={
                [
                  [
                    {
                      "alignSelf": "stretch",
                      "borderBottomColor": "white",
                      "borderBottomWidth": 1,
                      "flexDirection": "column",
                      "justifyContent": "space-between",
                    },
                    {
                      "borderColor": "white",
                      "borderWidth": 1,
                      "paddingHorizontal": 8,
                      "paddingVertical": 12,
                    },
                  ],
                  false,
                  {
                    "borderColor": "white",
                  },
                  {
                    "borderBottomColor": "white",
                  },
                ]
              }
            >
              <Text
                style={
                  [
                    [
                      {
                        "color": "#6D6D6D",
                        "fontSize": 14,
                        "paddingHorizontal": 4,
                      },
                      undefined,
                    ],
                    false,
                    {},
                  ]
                }
              >
                Positions
              </Text>
              <TextInput
                autoCapitalize="none"
                autoCorrect={false}
                editable={true}
                maxLength={300}
                multiline={true}
                onBlur={[Function]}
                onChangeText={[Function]}
                onFocus={[Function]}
                placeholder="What were your titles in the past?"
                placeholderTextColor="#6D6D6D"
                secureTextEntry={false}
                style={
                  [
                    [
                      {
                        "color": "white",
                        "fontSize": 14,
                        "paddingHorizontal": 10,
                        "paddingVertical": 10,
                      },
                      {
                        "color": "white",
                        "marginTop": 12,
                      },
                    ],
                    false,
                  ]
                }
                value=""
              />
            </View>
            <View
              style={
                {
                  "flex": 1,
                  "flexDirection": "row",
                  "justifyContent": "space-between",
                }
              }
            >
              <Text
                style={
                  [
                    [
                      {
                        "color": "#FF002E",
                        "fontSize": 12,
                        "marginTop": 8,
                        "paddingHorizontal": 10,
                      },
                      undefined,
                    ],
                    {
                      "color": "#FF002E",
                      "flex": 1,
                    },
                  ]
                }
              />
              <Text
                style={
                  [
                    {
                      "color": "white",
                      "fontSize": 12,
                      "marginTop": 8,
                      "paddingHorizontal": 10,
                    },
                    undefined,
                  ]
                }
              >
                0/300
              </Text>
            </View>
          </View>
          <View
            style={
              [
                {
                  "alignSelf": "stretch",
                  "flexDirection": "column",
                },
                {
                  "marginTop": 18,
                },
              ]
            }
            testID="container-id"
          >
            <View
              style={
                [
                  [
                    {
                      "alignSelf": "stretch",
                      "borderBottomColor": "white",
                      "borderBottomWidth": 1,
                      "flexDirection": "column",
                      "justifyContent": "space-between",
                    },
                    {
                      "borderColor": "white",
                      "borderWidth": 1,
                      "paddingHorizontal": 8,
                      "paddingVertical": 12,
                    },
                  ],
                  false,
                  {
                    "borderColor": "white",
                  },
                  {
                    "borderBottomColor": "white",
                  },
                ]
              }
            >
              <Text
                style={
                  [
                    [
                      {
                        "color": "#6D6D6D",
                        "fontSize": 14,
                        "paddingHorizontal": 4,
                      },
                      undefined,
                    ],
                    false,
                    {},
                  ]
                }
              >
                Speakings
              </Text>
              <TextInput
                autoCapitalize="none"
                autoCorrect={false}
                editable={true}
                maxLength={300}
                multiline={true}
                onBlur={[Function]}
                onChangeText={[Function]}
                onFocus={[Function]}
                placeholder="What have you spoke about in the past?"
                placeholderTextColor="#6D6D6D"
                secureTextEntry={false}
                style={
                  [
                    [
                      {
                        "color": "white",
                        "fontSize": 14,
                        "paddingHorizontal": 10,
                        "paddingVertical": 10,
                      },
                      {
                        "color": "white",
                        "marginTop": 12,
                      },
                    ],
                    false,
                  ]
                }
                value=""
              />
            </View>
            <View
              style={
                {
                  "flex": 1,
                  "flexDirection": "row",
                  "justifyContent": "space-between",
                }
              }
            >
              <Text
                style={
                  [
                    [
                      {
                        "color": "#FF002E",
                        "fontSize": 12,
                        "marginTop": 8,
                        "paddingHorizontal": 10,
                      },
                      undefined,
                    ],
                    {
                      "color": "#FF002E",
                      "flex": 1,
                    },
                  ]
                }
              />
              <Text
                style={
                  [
                    {
                      "color": "white",
                      "fontSize": 12,
                      "marginTop": 8,
                      "paddingHorizontal": 10,
                    },
                    undefined,
                  ]
                }
              >
                0/300
              </Text>
            </View>
          </View>
          <View
            style={
              [
                {
                  "alignSelf": "stretch",
                  "flexDirection": "column",
                },
                {
                  "marginTop": 18,
                },
              ]
            }
            testID="container-id"
          >
            <View
              style={
                [
                  [
                    {
                      "alignSelf": "stretch",
                      "borderBottomColor": "white",
                      "borderBottomWidth": 1,
                      "flexDirection": "column",
                      "justifyContent": "space-between",
                    },
                    {
                      "borderColor": "white",
                      "borderWidth": 1,
                      "paddingHorizontal": 8,
                      "paddingVertical": 12,
                    },
                  ],
                  false,
                  {
                    "borderColor": "white",
                  },
                  {
                    "borderBottomColor": "white",
                  },
                ]
              }
            >
              <Text
                style={
                  [
                    [
                      {
                        "color": "#6D6D6D",
                        "fontSize": 14,
                        "paddingHorizontal": 4,
                      },
                      undefined,
                    ],
                    false,
                    {},
                  ]
                }
              >
                Contributions
              </Text>
              <TextInput
                autoCapitalize="none"
                autoCorrect={false}
                editable={true}
                maxLength={300}
                multiline={true}
                onBlur={[Function]}
                onChangeText={[Function]}
                onFocus={[Function]}
                placeholder="What were your contributions?"
                placeholderTextColor="#6D6D6D"
                secureTextEntry={false}
                style={
                  [
                    [
                      {
                        "color": "white",
                        "fontSize": 14,
                        "paddingHorizontal": 10,
                        "paddingVertical": 10,
                      },
                      {
                        "color": "white",
                        "marginTop": 12,
                      },
                    ],
                    false,
                  ]
                }
                value=""
              />
            </View>
            <View
              style={
                {
                  "flex": 1,
                  "flexDirection": "row",
                  "justifyContent": "space-between",
                }
              }
            >
              <Text
                style={
                  [
                    [
                      {
                        "color": "#FF002E",
                        "fontSize": 12,
                        "marginTop": 8,
                        "paddingHorizontal": 10,
                      },
                      undefined,
                    ],
                    {
                      "color": "#FF002E",
                      "flex": 1,
                    },
                  ]
                }
              />
              <Text
                style={
                  [
                    {
                      "color": "white",
                      "fontSize": 12,
                      "marginTop": 8,
                      "paddingHorizontal": 10,
                    },
                    undefined,
                  ]
                }
              >
                0/300
              </Text>
            </View>
          </View>
          <View
            style={
              [
                {
                  "alignItems": "center",
                  "alignSelf": "stretch",
                  "flexBasis": 0,
                  "flexDirection": "row",
                  "flexGrow": 1,
                  "flexShrink": 1,
                  "height": 60,
                  "justifyContent": "space-between",
                  "marginBottom": 48,
                  "marginTop": 36,
                },
                undefined,
              ]
            }
          >
            <View
              accessible={true}
              collapsable={false}
              focusable={true}
              onClick={[Function]}
              onResponderGrant={[Function]}
              onResponderMove={[Function]}
              onResponderRelease={[Function]}
              onResponderTerminate={[Function]}
              onResponderTerminationRequest={[Function]}
              onStartShouldSetResponder={[Function]}
              style={
                {
                  "alignSelf": "stretch",
                  "flex": 1,
                  "opacity": 1,
                }
              }
              testID="button-update"
            />
          </View>
        </View>
      </View>
    </RCTScrollView>
  </View>
</RNCSafeAreaProvider>
`;
